File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x: x[1])
    camera=-30001
    count=0
    for start, end in routes:
        if camera < start:
            count+=1
            camera=end
    return count

-20 -15
-18 -13
-14 -5
-5  -3

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 0;
    int camera = -30001;
    sort(routes.begin(), routes.end(), [](const vector<int>& a,const vector<int>& b){return a[1]<b[1];});
    for(auto v: routes){
        int s=v[0];
        int e=v[1];
        if(camera<s){
            answer++;
            camera = e;
        }
    }
    return answer;
}